### PR TITLE
Refactored ParseACL to use POJOs instead of JSON objects

### PR DESCRIPTION
**Pull request by lukas1994**

### CODE REVIEW

**Comment by lukas1994** [comment]
The new ```Permissions``` class got quite big, should we put it in a new file @grantland ?

File: Parse/src/main/java/com/parse/ParseACL.java [hunk 1]
```diff
@@ -60,21 +61,63 @@ public static void setDefaultACL(ParseACL acl, boolean withAccessForCurrentUser)
...
+    /* package */ static Permissions createPermissionsFromJSONObject(JSONObject object) {
+      boolean read = object.optBoolean(READ_PERMISSION, false);
+      boolean write = object.optBoolean(WRITE_PERMISSION, false);
+      return new Permissions(read, write);
+    }
+
+    /* package */ boolean isEmpty() {
+      return !(read || write);
+    }
+  }
```
Comment (grantland) [inline Parse/src/main/java/com/parse/ParseACL.java]: Could we move this static class either above the instance variables or at the bottom of this class?

**Comment by grantland** [comment]
The `Permissions` class seems fine, esp since the abstraction is isolated to `ParseACL`.

I do think it might be worthwhile to make `Permissions` immutable, sorry I didn't have this comment in your initial inquiry. Thoughts @hallucinogen?

**Comment by lukas1994** [comment]
Yeah that's a good idea. I'll make it immutable.

**Comment by grantland** [comment]
a nit a small optimization

**Comment by grantland** [comment]
LGTM, please squash and I can merge!

**Pull request merged**